conda/repodata.py

```diff
@@ -0,0 +1,99 @@
+# (c) Continuum Analytics, Inc. / http://continuum.io
+# All Rights Reserved
+#
+# conda is distributed under the terms of the BSD 3-clause license.
+# Consult LICENSE.txt or http://opensource.org/licenses/BSD-3-Clause.
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+from .core import repodata as _impl
+from .connection import CondaSession
+from os.path import join
+
+
+class RepoData(object):
+    """This object represents all the package metainfo of a single channel."""
+
+    _data = {}
+
+    @staticmethod
+    def enable(url, name, priority, cache_dir=None):
+        RepoData._data[url] = RepoData(url, name, priority, cache_dir)
+
+    @staticmethod
+    def get(url):
+        return RepoData._data[url]
+
+    @staticmethod
+    def clean():
+        RepoData._data = {}
+
+    @staticmethod
+    def sync_all(use_cache=False):
+        try:
+            import concurrent.futures
+            with concurrent.futures.ThreadPoolExecutor(10) as e:
+                for rd in RepoData._data.values():
+                    e.submit(rd.sync(use_cache=use_cache, session=CondaSession()))
+        except (ImportError) as e:
+            for rd in RepoData._data.values():
+                rd.sync(use_cache=use_cache, session=CondaSession())
+
+    def __init__(self, url, name, priority, cache_dir=None):
```

Comment:
Sure, I'm right now looking into a RepoData metaclass that helps with some of that. I think I can get rid of the 'priority' by using an ordereddict. I'm not sure where the 'name' is used later in the code. It may be a convenience for users that doesn't have any use once the RepoData is instantiated.